exporters/jaeger/src/thrift_sender.cc

```diff
@@ -0,0 +1,117 @@
+// Copyright 2021, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "thrift_sender.h"
+#include "udp_transport.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter
+{
+namespace jaeger
+{
+
+using namespace jaegertracing;
+
+ThriftSender::ThriftSender(std::unique_ptr<Transport> &&transport)
+    : transport_(std::move(transport)),
+      protocol_factory_(new apache::thrift::protocol::TCompactProtocolFactory()),
+      thrift_buffer_(new apache::thrift::transport::TMemoryBuffer())
+{}
+
+int ThriftSender::Append(std::unique_ptr<Recordable> &&span) noexcept
+{
+  if (span == nullptr)
+  {
+    return 0;
+  }
+
+  std::lock_guard<std::mutex> guard{lock_};
+
+  uint32_t max_span_bytes = transport_->MaxPacketSize() - kEmitBatchOverhead;
+  if (process_.serviceName.empty())
+  {
+    // TODO: populate Span.Process from OpenTelemetry resources.
+    // process_bytes_size_ = CalcSizeOfSerializedThrift(process_);
+    // max_span_bytes -= process_byte_size_;
+    process_.serviceName = "OTel-Jaeger";
+
+    process_bytes_size_ = CalcSizeOfSerializedThrift(process_);
+    max_span_bytes -= process_bytes_size_;
+  }
+
+  thrift::Span &jaeger_span = *span->Span();
+  jaeger_span.__set_tags(span->Tags());
+
+  const uint32_t span_size = CalcSizeOfSerializedThrift(jaeger_span);
+  if (span_size > max_span_bytes)
+  {
+    // TODO, log too large span error.
```

Comment:
This would be nice to have, but how does otel-cpp log in general? I.e. I don't think it should write to stdout or stderr

Reply:
Yes, let's unify the SDK internal logging once we have it.